src/app/game-state/log.service.ts

```diff
@@ -63,10 +70,7 @@ export class LogService {
 
     if (log.length === 0 || ((newMessage.timestamp - log[0].timestamp) > LOG_MERGE_INTERVAL_MS) || !log[0].message.includes(newMessage.message)) {
       // Initialization || Repeat Not Found || Repeat is not within 1 second
-      log.unshift(newMessage);
-      if (this.logTopics.includes(topic)) {
-        this.addToCurrentLog(newMessage);
-      }
+      log.push(newMessage);
```

Comment:
Ripped out addtoCurrentLog since it was pushing one at a time and just did a full sort and slice.